app/site.rb

```diff
@@ -117,6 +117,12 @@ def git?
 			cache_path.join('.git').directory?
 		end
 
+		def log
+			git.log.tap(&:size)
```

Comment:
`git.log` is always an abstraction around the idea of the "git log".  It is lazy, so just asking for the log does not actually even check to see if there is a git repo at all.  That is why I have to ask it for something (like the elements, or here the size) in order to force it to look in the repo and thus throw the exception if things go poorly.